src/Artemis.Core/Models/Profile/Profile.cs

```diff
@@ -97,6 +103,13 @@ public override void Update(double deltaTime)
 
             foreach (ProfileScript profileScript in Scripts)
                 profileScript.OnProfileUpdated(deltaTime);
+
+            const double OPACITY_PER_SECOND = 1;
+            
+            if (ShouldBeEnabled && Opacity < 1)
+                Opacity = Math.Clamp(Opacity + OPACITY_PER_SECOND * deltaTime, 0d, 1d);
+            if (!ShouldBeEnabled && Opacity > 0)
+                Opacity = Math.Clamp(Opacity - OPACITY_PER_SECOND * deltaTime, 0d, 1d);
```

Comment:
Is there any downside in changing this to something like this to allow users set the smoothing time?


` const double TRANSITION_TIME = 3d;

            if (ShouldBeEnabled && Opacity < 1)
                Opacity = Math.Clamp(Opacity + deltaTime / TRANSITION_TIME, 0d, 1d);
            if (!ShouldBeEnabled && Opacity > 0)
                Opacity = Math.Clamp(Opacity - deltaTime / TRANSITION_TIME, 0d, 1d);`

Reply:
Me and Robert decided on not including time and easing method because it felt like too much detail. You can do whatever you want within the layer via start and end segments. This is just meant as a quick way to get the effect without needing to make it manually.